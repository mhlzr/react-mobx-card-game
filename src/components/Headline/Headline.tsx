import React, { ReactElement, FunctionComponent, ReactNode } from 'react';
import styled from 'styled-components';

const HeadlineLayout = styled.h1`
    font-size: 2em;
    font-weight: normal;
    margin: 0;
`;

interface HeadlineProps {
    children: ReactNode;
}

export const Headline: FunctionComponent<HeadlineProps> = ({ children }: HeadlineProps): ReactElement => {
    return (
        <HeadlineLayout>
            {children}
        </HeadlineLayout>
    )
}
