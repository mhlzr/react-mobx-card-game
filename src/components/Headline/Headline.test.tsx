import React from 'react';
import { shallow } from 'enzyme'

import { Headline } from './Headline';

describe('Headline', () => {

    it('renders with a label if one is passed', () => {
        const wrapper = shallow(<Headline>Test</Headline>);
        expect(wrapper.text()).toEqual('Test');
    });

    it('renders with child-components', () => {
        const wrapper = shallow(<Headline><span>Child</span></Headline>);
        expect(wrapper.find('span')).toHaveLength(1);
        expect(wrapper.find('span').text()).toEqual('Child');
    });

    it('has a snapshot', () => {
        const wrapper = shallow(<Headline>Test</Headline>);
        expect(wrapper).toMatchSnapshot();
    });
})
