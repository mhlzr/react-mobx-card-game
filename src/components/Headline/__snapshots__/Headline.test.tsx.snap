// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline has a snapshot 1`] = `
<styled.h1>
  Test
</styled.h1>
`;
